src/helpers/binding.ts

```diff
@@ -0,0 +1,163 @@
+const placeholderRegExp =
```

Comment:
While you can probably just skip this for the purposes of this review... there is a lot of pretty cool magic in here. I mean, look at that regex! But truly, this module simply parses SQL statements, and does its best to do client-side binding of parameters. It was a lot of work, so please at least appreciate it!

Soonish, this can be dropped in favor of real prepared statements. Though quite a bit of the auto-encoding stuff will still be relevant. For instance, this does its best to encode things like floats, numbers, dates, and binary data into the right format for our SQL spec. For now, I allow floats to be passed through, but this will be checked by our parser so it should be safe.

Reply:
Well, ideally no one is reading this as it isn't part of the public API. But yeh good call. I'll add a comment.